### PR TITLE
Adding Convection-Diffusion-Production Finite Elements

### PR DESCRIPTION
**Pull request by TheoChristo**

### CODE REVIEW

**Comment by TheoChristo** [comment]
Need to update code to use MGroup.Constitutive as a nuget (instead of project reference), as soon as the latter is created from my latest pull request at https://github.com/mgroupntua/Constitutive

**Comment by TheoChristo** [comment]
Updated code to remove unnecessary comments and reference Constitutive.ConvectionDiffusion as nuget

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+		public Matrix BuildShapeFunctionMatrix(double[] shapeFunctions)
+		{
+			return Matrix.CreateFromArray(shapeFunctions, 1, shapeFunctions.Length);
+		}
+
+		public IReadOnlyList<IReadOnlyList<IDofType>> GetElementDofTypes() => dofTypes;
+
+		public void ResetConstitutiveLawModified()
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:253]: Leave method blank

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public Tuple<double[], double[]> CalculateResponse(double[] localDisplacements)
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:258]: Method should be implemented with constitutive matrix

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public double[] CalculateResponseIntegral()
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:263]: Method should be implemented with constitutive matrix

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public double[] CalculateResponseIntegralForLogging(double[] localDisplacements)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void SaveConstitutiveLawState()
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:273]: Leave method blank

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public void ClearConstitutiveLawState()
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:278]: Leave method blank

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public void ClearConstitutiveLawStresses()
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:283]: Leave method blank

File: src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		public double[] CalculateResponseIntegral()
+		{
+			throw new NotImplementedException();
+		}
+
+		public double[] CalculateResponseIntegralForLogging(double[] localDisplacements)
+		{
+			throw new NotImplementedException();
```
Comment (georgestavroulakis) [inline src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs:268]: Method should be implemented as with CalculateResponseIntegral but with logging (check Constitutive)

**Review by GoatOfRafin — CHANGES_REQUESTED** [review]
Please check constitutive-law oriented methods for implementation

**Pull request merged**